File: src/translate-app/frontend/src/types/file.ts
/** Giới hạn số trang cho một lần dịch file — đồng bộ BE `internal/controller/file/translate.go` (`maxFilePages`). */
export const MAX_FILE_PAGE_COUNT = 200

export interface FileAttachment {
  id: string
  sessionId: string
  fileName: string
  fileType: 'pdf' | 'docx'
  fileSize: number
  originalPath: string
  sourcePath: string
  translatedPath: string
  pageCount: number
  charCount: number
  style: 'casual' | 'business' | 'academic'
  modelUsed: string
  status: 'pending' | 'processing' | 'done' | 'error'
  errorMsg: string
  createdAt: string
  updatedAt: string
}

export interface FileResult {
  fileId: string
  fileName: string
  fileType: 'pdf' | 'docx'
  charCount: number
  pageCount: number
  tokensUsed: number
}

export interface FileProgress {
  chunk: number
  total: number
  percent: number
}

File: src/translate-app/backend/internal/model/file.go
package model

type File struct {
	ID             string           `json:"id"`
	SessionID      string           `json:"sessionId"`
	FileName       string           `json:"fileName"`
	FileType       string           `json:"fileType"`
	FileSize       int64            `json:"fileSize"`
	OriginalPath   string           `json:"originalPath"`
	SourcePath     string           `json:"sourcePath"`
	TranslatedPath string           `json:"translatedPath"`
	PageCount      int              `json:"pageCount"`
	CharCount      int              `json:"charCount"`
	Style          TranslationStyle `json:"style"`
	ModelUsed      string           `json:"modelUsed"`
	Status         string           `json:"status"`
	ErrorMsg       string           `json:"errorMsg"`
	CreatedAt      string           `json:"createdAt"` // RFC3339
	UpdatedAt      string           `json:"updatedAt"`
}
